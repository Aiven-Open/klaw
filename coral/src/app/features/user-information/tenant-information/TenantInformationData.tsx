import { Box, Typography } from "@aivenio/aquarium";

interface TenantInformationDataProps {
  tenantName: string;
  orgName: string;
  contactPerson: string;
  description: string;
}

const TenantInformationData = ({
  tenantName,
  orgName,
  contactPerson,
  description,
}: TenantInformationDataProps) => {
  return (
    <Box.Flex flexDirection="column" gap={"l2"}>
      <Box.Flex flexDirection="column">
        <Typography.SmallStrong htmlTag="dt">
          Description
        </Typography.SmallStrong>
        <Typography.Default htmlTag="dd">{description}</Typography.Default>
      </Box.Flex>
      <Box.Flex flexDirection="column">
        <Typography.SmallStrong htmlTag="dt">
          Organization
        </Typography.SmallStrong>
        <Typography.Default htmlTag="dd">{orgName}</Typography.Default>
      </Box.Flex>
      <Box.Flex flexDirection="column">
        <Typography.SmallStrong htmlTag="dt">Tenant</Typography.SmallStrong>
        <Typography.Default htmlTag="dd">{tenantName}</Typography.Default>
      </Box.Flex>
      <Box.Flex flexDirection="column">
        <Typography.SmallStrong htmlTag="dt">
          Contact person
        </Typography.SmallStrong>
        <Typography.Default htmlTag="dd">{contactPerson}</Typography.Default>
      </Box.Flex>
    </Box.Flex>
  );
};

export { TenantInformationData };
