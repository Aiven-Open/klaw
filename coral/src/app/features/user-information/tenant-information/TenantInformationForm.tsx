const TenantInformationForm = () => {
  return <form role="form">To be added later</form>;
};

export { TenantInformationForm };
