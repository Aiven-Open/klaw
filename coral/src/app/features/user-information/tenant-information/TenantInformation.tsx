import { Alert, Box, Input } from "@aivenio/aquarium";
import { useQuery } from "@tanstack/react-query";
import { TenantInformationData } from "src/app/features/user-information/tenant-information/TenantInformationData";
import { getMyTenantInfo } from "src/domain/user/user-api";
import { parseErrorMsg } from "src/services/mutation-utils";

const TenantInformation = () => {
  const { data, isLoading, isError, error } = useQuery(
    ["getMyTenantInfo"],
    getMyTenantInfo
  );

  if (isLoading) {
    return (
      <Box.Flex flexDirection="column" maxWidth={"lg"}>
        <Input.Skeleton />
        <Input.Skeleton />
        <Input.Skeleton />
      </Box.Flex>
    );
  }

  if (isError) {
    return <Alert type="error">{parseErrorMsg(error)}</Alert>;
  }

  return (
    <TenantInformationData
      tenantName={data.tenantName}
      orgName={data.orgName}
      contactPerson={data.contactPerson}
      description={data.tenantDesc}
    />
  );
};

export { TenantInformation };
