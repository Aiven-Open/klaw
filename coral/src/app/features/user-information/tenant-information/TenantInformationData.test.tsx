import { cleanup, render } from "@testing-library/react";
import { TenantInformationData } from "src/app/features/user-information/tenant-information/TenantInformationData";

const props = {
  tenantName: "default",
  contactPerson: "Klaw Administrator",
  orgName: "Default Organization",
  description: "This is the default description",
};

describe("TenantInformationData.tsx", () => {
  afterAll(cleanup);

  describe("renders correct DOM with provided props", () => {
    it("renders correct DOM with required props", () => {
      render(<TenantInformationData {...props} />);

      expect(screen).toMatchSnapshot();
    });
  });
});
