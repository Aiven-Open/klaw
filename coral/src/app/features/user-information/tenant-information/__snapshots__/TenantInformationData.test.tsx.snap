// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenantInformationData.tsx renders correct DOM with provided props renders correct DOM with required props 1`] = `Screen {}`;
